### PR TITLE
Retrain huwiki models

### PR DESCRIPTION
**Pull request by accraze**
Retraining the damaging and goodfaith models for huwiki based off of a recent labeling campaign: https://labels.wmflabs.org/campaigns/huwiki/92/

This also includes an update to `merge_labels` utility, which will now handle revisions with an empty `autolabel` field. In the case that this field is empty, we assume the worst case which is `"needs_review": True`

### CODE REVIEW

**Comment by accraze** [comment]
@halfak I'm getting an error when I try to create the new `labeled_revisions` data

```
make datasets/huwiki.labeled_revisions.json
./utility merge_labels datasets/huwiki.badfaith_relabeling_revisions.500_2019.json datasets/huwiki.original_labeled_revisions.40k_2016.json > datasets/huwiki.labeled_revisions.json
Traceback (most recent call last):
  File "./utility", line 4, in <module>
    editquality.main()
  File "/srv/home/accraze/editquality/editquality/editquality.py", line 50, in main
    module.main(sys.argv[2:])
  File "/srv/home/accraze/editquality/editquality/utilities/merge_labels.py", line 73, in main
    jsonlines = [json.dumps(l) for l in merged_labels]
  File "/srv/home/accraze/editquality/editquality/utilities/merge_labels.py", line 106, in run
    if not revision['autolabel']['needs_review']:
KeyError: 'needs_review'
```
The new data looks fine, but when I look at `datasets/huwiki.original_labeled_revisions.40k_2016.json`, I see that each revision looks similar to this:

```
{"goodfaith": true, "auto_labeled": false,  "autolabel": {}, "damaging": false, "rev_id": 16416637,}
```
(note the `"autolabel":{}` )

Am I forgetting a step? Or maybe an incorrect config?

**Comment by halfak** [comment]
They have an autolabel field when they get downloaded.  You can see that here: https://labels.wmflabs.org/campaigns/huwiki/92/?tasks

Still digging...

**Comment by halfak** [comment]
When I make the badfaith relabeling dataset (should probably change that name to "badfaith_or_damaging" to avoid confusion), it includes an autolabel field. 

```
{"goodfaith": false, "auto_labeled": false, "reverted_for_damage": false, "autolabel": {"review_reason": null, "needs_review": true}, "damaging": true, "rev_id": 17276091}
```

**Comment by accraze** [comment]
@halfak yeah the badfaith dataset looks fine, but the `autolabel` field is empty on the original dataset (`datasets/huwiki.original_labeled_revisions.40k_2016.json`). I get the error when I try to merge the two of them together. Should I even be merging the old and new labels together?

**Comment by halfak** [comment]
OK I think I have figured out what the source of this is.  So very few of the records are missing autolabel.  It looks like it is coming from observations stored in Wikilabels that are not appearing in the autolabel dataset.  This is because the articles the revisions belong to have been deleted.  

Here's an example of such a record: 
```
{"auto_labeled": false, "rev_id": 16318074, "goodfaith": true, "damaging": true, "autolabel": {}}
```

Here's the diff that the revision points to: 
https://hu.wikipedia.org/wiki/?diff=16318074

Here's what I think that we should do:
1. Update the merge_labels utility so that it doesn't error when that field isn't present.  Instead it should just assume the worst (`"needs review": true`).
2. Re-run this model pipeline.  
3. Victory.

File: config/wikis/huwiki.yaml [hunk 1]
```diff
@@ -29,39 +31,44 @@ balanced_5k_samples:
...
     - autolabeled_revisions.40k_2016
+  labeled_revisions.40k_2019:
+    - badfaith_or_damaging_relabeling_revisions.5k_2019
+    - original_labeled_revisions.40k_2016
 
 extracted_samples:
-  labeled_revisions.w_cache.40k_2016:
-    sample: labeled_revisions.40k_2016
+  labeled_revisions.w_cache.40k_2019:
+    sample: labeled_revisions.40k_2019
```
Comment (halfak) [inline config/wikis/huwiki.yaml]: Should still be 2016 as the underlying data is from 2016.

File: editquality/utilities/merge_labels.py [hunk 2]
```diff
@@ -178,3 +178,11 @@ def has_all_labels(revision, labels):
...
             return False
     return True
+
+
+def needs_review(revision):
+    """
+    Returns bool if revision needs review.
+    If autolabel is empty, assume true.
+    """
+    return revision['autolabel'].get('needs_review', True)
```
Comment (halfak) [inline editquality/utilities/merge_labels.py:188]: I like it.

File: editquality/utilities/tests/test_merge_labels.py [hunk 3]
```diff
@@ -147,3 +147,19 @@ def test_merge_labels(human, auto, expected):
...
+    res = merge_labels.needs_review(revision)
+    assert res == True  # noqa
+    #  check false
+    revision = {'autolabel': {'needs_review': False}}
+    res = merge_labels.needs_review(revision)
+    assert res == False  # noqa
+    #  check true
+    revision['autolabel']['needs_review'] = True
+    res = merge_labels.needs_review(revision)
+    assert res == True  # noqa
```
Comment (halfak) [inline editquality/utilities/tests/test_merge_labels.py:165]: Nice work including a set of tests.

**Review by halfak — CHANGES_REQUESTED** [review]
Just one minor change in data file naming.  Otherwise this looks good.

**Pull request merged**